cpp/src/IceLocatorDiscovery/PluginI.cpp

```diff
@@ -94,9 +96,9 @@ class LocatorI : public Ice::BlobjectArrayAsync,
 
     string _instanceName;
     bool _warned;
-    Ice::LocatorPrxPtr _locator;
-    map<string, Ice::LocatorPrxPtr> _locators;
-    Ice::LocatorPrxPtr _voidLocator;
+    optional<Ice::LocatorPrx> _locator;
```

Comment:
Not clear if we can move the initialization of `_locator` to the PluginI constructor to avoid the optional.